StyleCop.Analyzers/StyleCop.Analyzers.Test/ReadabilityRules/SA1108UnitTests.cs

```diff
@@ -0,0 +1,434 @@
+﻿namespace StyleCop.Analyzers.Test.ReadabilityRules
+{
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.CodeAnalysis.Diagnostics;
+    using StyleCop.Analyzers.ReadabilityRules;
+    using TestHelper;
+    using Xunit;
+
+    public class SA1108UnitTests : CodeFixVerifier
+    {
+        [Fact]
+        public async Task TestEmptySource()
+        {
+            var testCode = string.Empty;
+            await this.VerifyCSharpDiagnosticAsync(testCode, EmptyDiagnosticResults, CancellationToken.None);
+        }
+
+        [Fact]
+        public async Task TestIfStatementCommentBetweenStatementDeclarationAndOpeningCurlyBracket()
+        {
+            var testCode = @"
+public class Foo
+{
+    public void F()
+    {
+        //comment
+        if (true)
+        // Comment
+        {
+            //comment
+        }
+        else
+        //comment
+        {
+            //inner comment
+        }
+    }
+}";
+
+            var firstDiagnostic = this.CSharpDiagnostic().WithLocation(8, 9);
+            var secondDiagnostic = this.CSharpDiagnostic().WithLocation(13, 9);
+
+            await this.VerifyCSharpDiagnosticAsync(testCode, new [] {firstDiagnostic, secondDiagnostic}, CancellationToken.None);
```

Comment:
The spacing is off here. To follow stylecop it should be 
```csharp
new[] { firstDiagnostic, secondDiagnostic }
```
There are a few occurences in this file.